nightfall-optimist/src/app.mjs

```diff
@@ -12,8 +19,18 @@ import {
   debug,
 } from './routes/index.mjs';
 
+const spec = swaggerJsDoc(options);
+const environment = config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
+logger.debug(`********* Optimist environment vars ${JSON.stringify(environment)}`); // TODO review logs
+
 const app = express();
 
+const ethPrivateKey = environment.PROPOSER_KEY;
```

Comment:
Why are we reading proposer erh address and private key?

Reply:
The private key is needed for signing transactions.
The address is part of the transaction. Also, most of the contracts can filter results based on address.

Is this what you meant, or do you have other concerns, ie:

- Storing key in memory
- Similar to `API key`, changing private keys

Reply:
We could derive an eth address from the private key, is that your proposal @druiz0992 ? Sorry I don't understand what you mean

Reply:
This question is just to understand the reason. I got the impression from other files that you are retrieving key and address information from env variables, so I don’t understand why you also copy here

Reply:
Addressed in #1262